### ISSUE TITLE
GRPO reward hacking

### ISSUE THREAD

**Issue by Kidand** [huggingface/open-r1#407]
After attempting to train using the configuration of recipes/Qwen2.5-1.5B-Instruct/grpo, I found that the model tends to get stuck in a long-term local optimum early on and exhibits reward hacking after breaking out of this local optimum in the mid-stage. I would like to inquire if open-r1 has successfully trained a stable COT model using the current version of the code with grpo training.

**Comment by skepsun** [comment]
In my view, they are still making an effort. There are multiple factors that influence the trend of training:

1. System Prompt:

2. Format Reward Function

3. Hyperparameters: Such as num_generations, beta, lr, and so on. 

4. The Selection of Model: In my opinion, only the base model without any Supervised Fine-Tuning (SFT) is the suitable choice for reproducing r1-zero. Models like Qwen2.5-1.5B-Instruct may not be the most appropriate for this particular reproduction task.

**Comment by Wangjunhc** [comment]
> In my view, they are still making an effort. There are multiple factors that influence the trend of training:
> 
> 1. System Prompt:
> 2. Format Reward Function
> 3. Hyperparameters: Such as num_generations, beta, lr, and so on.
> 4. The Selection of Model: In my opinion, only the base model without any Supervised Fine-Tuning (SFT) is the suitable choice for reproducing r1-zero. Models like Qwen2.5-1.5B-Instruct may not be the most appropriate for this particular reproduction task.

Why do you think the model after SFT is not suitable for reproduction? Do you have any references? Thank you

**Comment by Xingxl2studious** [comment]
+1

**Comment by jackswl** [comment]
@skepsun please kindly clarify on your 4th point, why do you think that way?
